src/reducers/index.js

```diff
@@ -0,0 +1,64 @@
+const initialState = {
+  posts: [],
+  post: null,
+  isLoaded: false,
+  error: false,
+  isLoggedOn: false,
+  loginButtonValue: 'Login'
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case 'FETCH_POSTS_REQUEST':
+      return {
+        ...state
```

Comment:
What for? You will copy all state on every request. This can cause rerender of application.

Do you really need this?